lighthouse-cli/test/cli/bin-test.js

```diff
@@ -6,38 +6,16 @@
 
 import fs from 'fs';
 
-import * as td from 'testdouble';
-import jestMock from 'jest-mock';
-
+import * as bin from '../../bin.js';
 import {LH_ROOT} from '../../../root.js';
 import {readJson} from '../../../lighthouse-core/test/test-utils.js';
 
-const mockRunLighthouse = jestMock.fn();
-const mockGetFlags = jestMock.fn();
-const mockAskPermission = jestMock.fn();
-const mockSentryInit = jestMock.fn();
-const mockLoggerSetLevel = jestMock.fn();
-
-/** @type {import('../../bin.js')} */
-let bin;
-before(async () => {
-  await td.replaceEsm('../../run.js', {
-    runLighthouse: mockRunLighthouse,
-  });
-  await td.replaceEsm('../../cli-flags.js', {
-    getFlags: mockGetFlags,
-  });
-  await td.replaceEsm('../../sentry-prompt.js', {
-    askPermission: mockAskPermission,
-  });
-  await td.replaceEsm('../../../lighthouse-core/lib/sentry.js', {
-    Sentry: {
-      init: mockSentryInit,
-    },
-  });
-  await td.replaceEsm('lighthouse-logger', undefined, {setLevel: mockLoggerSetLevel});
-  bin = await import('../../bin.js');
-});
+/** @type {import('./bin-test.mocks.js').TestContext} */
+// @ts-expect-error
+const testContext = global.lighthouseTestContext;
```

Comment:
This seems like a roundabout way of doing `import {testContext} from './bin-test.mocks.js'` as the first import of the file. Could we do that instead?

Eslint could be updated to enforce this import order.

Reply:
No, because the import cache is refreshed after every call to td.replaceEsm, hence this whole setup.

Reply:
^that's old knowledge. now, it's "source defined order" except for when there is async code in the root module scope, which all of these setup files have.